day_01/day_01.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
+import argparse
+import pathlib
+import re
+
+import day_01
+
+DEFAULT_DATA = pathlib.Path(day_01.__file__).parent.parent / "data"/ "day1.txt"
+
+
+def load(filename: str | pathlib.Path) -> list[str]:
+    with open(filename) as f:
+        return f.readlines()
+
+
+def solve_q1(data: list[str]) -> int:
+    digits: list[int] = []
+    digit_re = re.compile(r'\d')
```

Comment:
Compile the regex ahead of iteration - this will create the object once instead of re-creating it for every line of the file.